### ISSUE TITLE
one field result depend on other field result in graphql query result

### ISSUE THREAD

**Issue by pli2014** [graphql-java/graphql-java#1406]
```graphql
query graphql {
   hello:helloWorld(id:123){
    name
    age
  }
  aliasName:$hello.name
}
```
By this graphql.execution.ExecutionStrategy, Normally the executor can execute the entries in a grouped field set in whatever order it chooses. 
Actually, some data always is relative or dependent, which involved to path mapping or function operation. 

```
suppose hello.name="xxx"; 
aliasName:$hello.name   equal to  aliasName:"xxx"
aliasName: @js(function:"function(x,y){return x+y}", params:[$hello.name,"###"])  
equal to aliasName:"xxx###"
```
So I want to know what you think, how to enable it.

**Comment by andimarek** [comment]
I am sorry, but the functionality that you describe is not supported: you can not reference the return value of some fields in another field.

**Comment by bbakerman** [comment]
You cant do this.  Even if the field ordering was "garunteed" (which it is not and wont be).. there is no way from a data field for a field to gain access to the value of another field.

Also you have invented new "syntax" here of allowing access ($hello.name) into graphql which would just fail.   A field can have an alias but it MUST be a field.

This idea has ALL sorts of "variable scoping" challenges eg

```
{
     person {
         name
         friends {
             name
             person {
                  name
                  $person.name   # which scope??

```

In short I can see how you can do this AND we would not change graphql to allow a data fetcher to get previous results

**Comment by pli2014** [comment]
yes,  I think graphql is lack of some function capacity with JavaScript,  because we want to unify all data process(e.g. JS) in graphql. function + DSL = graphql which is suitable for querying data and processing data.
from async datafetcher, we need to async completable result for next reference of node.
